src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/AutoplayVuap.java

```diff
@@ -0,0 +1,285 @@
+    closeButtonSelector = By.cssSelector(String.format(CLOSE_BUTTON_SELECTOR_FORMAT, slot));
+    progressBarSelector = By.cssSelector(String.format(CURRENT_TIME_SELECTOR_FORMAT, slot));
+    replayOverlaySelector = By.cssSelector(String.format(REPLAY_BUTTON_SELECTOR_FORMAT, slot));
+  }
+
+  public void mute() {
+    if (!isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
+    }
+  }
+
+  public void unmute() {
+    if (isMuted()) {
+      wait.forElementClickable(speakerSelector).click();
+    }
+  }
+
+  public void togglePause() {
+    wait.forElementClickable(pauseOverlaySelector);
+    // hack to click pause overlay on mobile
+    clickByJS(pauseOverlaySelector);
+  }
+
+  public void play() {
+    if (isPausedWithOverlay()) {
+      togglePause();
+    } else if( !this.mobile ) {
+      clickOnArea(2);
+    } else {
+      clickOnArea(4);
+    }
+  }
+
+  private boolean isPausedWithOverlay() {
+    return isVideoPaused() && driver.findElement(pauseOverlaySelector).isDisplayed();
+  }
+
+  private Boolean isVideoPaused() {
+    return isDesktop() ? isDesktopVideoPaused() : isMobileVideoPaused();
+  }
+
+  private boolean isDesktop() {
+    return hasDesktopVideoElement();
+  }
+
+  private Boolean isMobileVideoPaused() {
+    return Boolean.valueOf(driver.findElement(getVideoSelector()).getAttribute("paused"));
+  }
+
+  private Boolean isDesktopVideoPaused() {
+    return Boolean.valueOf(usingImaBridge(webDriver -> {
+      final JavascriptActions js = new JavascriptActions();
+
+      return js.execute("document.querySelector('video') && document.querySelector('video').paused").toString();
+    }));
+  }
+
+  public void replay() {
+    wait.forElementVisible(replayOverlaySelector).click();
+  }
+
+  public void close() {
+    wait.forElementClickable(closeButtonSelector).click();
+  }
+
+  public void closeWithJS() {
+    clickByJS(closeButtonSelector);
+  }
+
+  private void clickByJS(By selector) {
+    new JavascriptActions(driver).execute("arguments[0].click();", driver.findElement(selector));
+  }
+
+  private void clickElementInsideAd(By selector) {
+    usingAdFrame(() -> {
+      // It need to be clicked by JS, because our templates elements covers each other
+      // and there is no way to click it by just .click()
+      clickByJS(selector);
+    });
+  }
+
+  public double getVideoHeightWhilePaused() {
+    return driver.findElement(pauseOverlaySelector).getSize().getHeight();
+  }
+
+  public double getAdSlotHeight() {
+    waitForAdToLoad();
+    return driver.findElement(By.id(slot)).getSize().getHeight();
+  }
+
+  public int getProgressBarWidth() {
+    return driver.findElement(progressBarSelector).getSize().getWidth();
+  }
+
+  public boolean hasStarted() {
+    return getCurrentTime() > 0;
+  }
+
+  public boolean isVisible() {
+    return isDesktop() ?
+           usingImaBridge(webDriver -> webDriver.findElement(By.tagName("video")).isDisplayed()) :
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "video" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)